pkg/costmodel/cluster.go

```diff
@@ -156,6 +157,7 @@ func ClusterDisks(client prometheus.Client, provider cloud.Provider, start, end
 	queryPVCost := fmt.Sprintf(`avg(avg_over_time(pv_hourly_cost[%s])) by (%s, persistentvolume,provider_id)`, durStr, env.GetPromClusterLabel())
 	queryPVSize := fmt.Sprintf(`avg(avg_over_time(kube_persistentvolume_capacity_bytes[%s])) by (%s, persistentvolume)`, durStr, env.GetPromClusterLabel())
 	queryActiveMins := fmt.Sprintf(`count(pv_hourly_cost) by (%s, persistentvolume)[%s:%dm]`, env.GetPromClusterLabel(), durStr, minsPerResolution)
+	queryPVStorageClass := fmt.Sprintf(`avg(avg_over_time(kube_persistentvolumeclaim_info[%s])) by (%s, volumename, storageclass)`, durStr, env.GetPromClusterLabel())
```

Comment:
I think we should be getting our information from all Persistent Volumes (PVs), not just Persistent Volume Claims (PVCs). What happens if the cluster has an unclaimed volume?

Reply:
We already emit the storage class for PVCs: https://github.com/opencost/opencost/blob/0094cba41fc0827e1598ceba9f6901739aed6386/pkg/metrics/pvcmetrics.go#L117

so it should be too hard to add that to our emission of data for PVs:
https://github.com/opencost/opencost/blob/0094cba41fc0827e1598ceba9f6901739aed6386/pkg/metrics/pvmetrics.go#L73

Reply:
Hmm, I see how this could be a problem. It looks like regular KSM currently has a `kube_persistentvolume_info` metric [1], which contains exactly the kind of data we're looking for. However, we haven't required this metric in the past so we would have to write a new emitter for it if we wanted to use it.

I'd like to call in some better metric knowledge here. @mbolt35 could you perhaps weigh in? I'm currently leaning towards adding a metric emitter for `kube_pv_info` but I'm not certain if its a good idea.

[1] https://github.com/kubernetes/kube-state-metrics/blob/master/docs/persistentvolume-metrics.md

Reply:
To answer question on unclaimed persistent volume, I created an unclaimed persistent volume by creating a persistent volume claim and not assign it to any pod, attached is the image, i dont think we print any information about unclaimed persistent volume. The different metrics used to compute cost and capacity for pv volume `pv_hourly_cost`, `kube_persistentvolume_capacity_bytes` doesnt have information about unclaimed PVs. 

The interesting thing is the metrics `kube_persistentvolumeclaim_info` actually has the data but since we never create a disk object for unclaimed pv it was never used attached is the image.

![Screen Shot 2022-08-08 at 3 53 56 PM](https://user-images.githubusercontent.com/11470561/183528079-98ef30c0-6d7b-4395-b1c5-e844059d4be4.png)

![Screen Shot 2022-08-08 at 3 48 07 PM](https://user-images.githubusercontent.com/11470561/183527810-5759c8ac-a864-4afa-9122-1f0e7c5553d5.png)

Reply:
This seems like the key decision for this PR. Talked with Bolt, and we agree with you Michael -- we should probably emit a new metric (something like `kubecost_pv_info` or similar) that has `storageclass` (and whatever else we want, but can start there).

We'd recommend putting that logic here: https://github.com/opencost/opencost/blob/develop/pkg/metrics/pvmetrics.go

And for examples, you can look to the KSM implementation of `kube_persistentvolume_info` here: https://github.com/kubernetes/kube-state-metrics/blob/7e28ed2722a8edfade41238ee3f0e0117ff041b6/internal/store/persistentvolume.go#L161-L249

Please let me know if I can help with that at all!

Sound good to you, @michaelmdresser?